### PR TITLE
Improve Generic driver doc

### PR DESCRIPTION
**Pull request by dgageot**
- Improve information about Password-protected SSH keys
- Add sudoer information
- Describe missing provisionning steps

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: docs/drivers/generic.md [hunk 1]
```diff
@@ -19,13 +19,48 @@ Machine to manage.
...
+which is the default behaviour when the `ssh` binary is available. If you're
+using the native client (with `--native-ssh`), using the SSH agent is not yet
+supported.
+
+    $ docker-machine create \
+      --driver generic \
+      --generic-ip-address=203.0.113.81 \
+      other
+
+### Sudo privileges
```
Comment (pdxjohnny) [inline docs/drivers/generic.md]: This resolves / removes the need for #1586 

Thanks!

**Comment by jeanlaurent** [comment]
LGTM

File: docs/drivers/generic.md (hunk 1 above)
Comment (simbamarufu1) [reply docs/drivers/generic.md:55]: This doesn't resolve the problem.

Comment (tad3j) [reply docs/drivers/generic.md:55]: Doesn't work for me either.

**Pull request merged**